pride-core/src/main/java/com/prezi/pride/Pride.java

```diff
@@ -165,7 +165,16 @@ public void removeModule(final String name) throws IOException {
 		final File moduleDir = getModuleDirectory(name);
 		logger.info("Removing " + name + " from " + moduleDir);
 		modules.remove(name);
-		FileUtils.deleteDirectory(moduleDir);
+		try {
+			FileUtils.deleteDirectory(moduleDir);
+		}
+		catch (IOException ex) {
+			// Found on Windows only, sometimes files cannot be deleted
+			// This can be due to background processes monitoring the filesystem or
+			// the user left a command shell open in the folder
+			logger.error("Removing " + name + " failed with error: " + ex.getMessage());
+			logger.error("Please remove folder " + moduleDir + " manually");
```

Comment:
I do agree with you that is would be better to change the exit value of the process. However, it is not clear to me how to propagate an exit value in this particular case.

From: Lóránt Pintér [mailto:notifications@github.com]
Sent: Monday, February 16, 2015 9:12 AM
To: prezi/pride
Cc: Coen Wouters
Subject: Re: [pride] Do not throw IOException when a module cannot be removed (#138)


In pride-core/src/main/java/com/prezi/pride/Pride.java<https://github.com/prezi/pride/pull/138#discussion_r24735949>:

> @@ -165,7 +165,16 @@ public void removeModule(final String name) throws IOException {

>              final File moduleDir = getModuleDirectory(name);

>              logger.info("Removing " + name + " from " + moduleDir);

>              modules.remove(name);

> -            FileUtils.deleteDirectory(moduleDir);

> +            try {

> +                    FileUtils.deleteDirectory(moduleDir);

> +            }

> +            catch (IOException ex) {

> +                    // Found on Windows only, sometimes files cannot be deleted

> +                    // This can be due to background processes monitoring the filesystem or

> +                    // the user left a command shell open in the folder

> +                    logger.error("Removing " + name + " failed with error: " + ex.getMessage());

> +                    logger.error("Please remove folder " + moduleDir + " manually");

I agree that we should not stop at the first delete error. But shouldn't this change the exit value of the process as well?

—
Reply to this email directly or view it on GitHub<https://github.com/prezi/pride/pull/138/files#r24735949>.